Add new dev cmds to Software CLI Reference guide (#751)
astronomer/docs: software/cli-reference.md

| `--webserver` | None       | Outputs only Webserver logs.                       |
| `--triggerer` | None       | Outputs only Triggerer logs.                       |

## astro dev parse

Parse the DAGs in a locally hosted Astro project to quickly check them for errors. For more information about testing DAGs locally, read [Run Tests with the Astro CLI](test-and-troubleshoot-locally.md#run-tests-with-the-astro-cli).

:::info

> This command requires Astro Runtime version `4.1.0`+ or for `pytest` to be present in your `requirments.txt` file. For more information, see [Astro Runtime Release Notes](https://docs.astronomer.io/astro/runtime-release-notes#astro-runtime-410).

Reviewer: ```suggestion
This command requires the [pytest](https://docs.pytest.org/en/7.1.x/contents.html) package. To install it to your Astro project, add `pytest` to your `requirements.txt` file.
```

@sunkickr Do we actually need `pytest` for this parse command? If so, I assume we want to scrap this Runtime note and just tell people to add `pytest`. And no expected issues on that with AC?

Reviewer: Looks like this requirement isn't listed for Astro: https://docs.astronomer.io/astro/cli-reference/astrocloud-dev-parse

Author: if you have the right runtime image you don't need pytest. if you don't have the right runtime image you do need pytest

Author: we added pytest to runtime so users would not need to add pytest to requirements

Reviewer 2: Does the CLI have error handling for this? Ie if someone runs the command before Runtime 4.1.0, the CLI tells them what went wrong and how to get `pytest`? If so, we should remove this note and always assume that a user is running latest.

`astro dev ps`

## astro dev pytest

> Run unit tests for your data pipelines on Astro with `pytest`, a testing framework for Python. When you run this command, the Astro CLI creates a local Python environment that includes your DAG code, dependencies, and Astro Runtime Docker image. The CLI then runs any pytests in the `tests` directory of your Astro project and shows you the results of those tests in your terminal.

Reviewer: ```suggestion
Run unit tests for your data pipelines with `pytest`, a testing framework for Python. When you run this command, the Astro CLI creates a local Python environment that includes your DAG code, dependencies, and Astronomer Certified Docker image. The CLI then runs any pytests in the `tests` directory of your Astro project and shows you the results of those tests in your terminal.
```

For more information on this functionality, see [Test and Troubleshoot Locally](test-and-troubleshoot-locally.md).

:::info

This command requires Astro Runtime version `4.1.0`+ or for `pytest` to be present in your `requirments.txt` file. For more information, see [Astro Runtime Release Notes](https://docs.astronomer.io/astro/runtime-release-notes#astro-runtime-410).

> :::

Reviewer: Same thing here

### Usage

> `astro dev pytest <pytest-filepath>` to run specfic Pytests on your DAGs. use `<pytest-filepath>` to specify a specific test. 

Reviewer: ```suggestion
`astro dev pytest <pytest-filepath>` to run specfic Pytests on your DAGs. Use `<pytest-filepath>` to specify a specific test. 
```

### Flags

| Flag          | Value Type | Usage                                              |
| ------------- | ---------- | -------------------------------------------------- |
| `--env`       | string     | The filepath to your environment variables. (The default is `.env`)  |

> 

Reviewer: Any reason these flags are different than in Astro CLI docs? I know it's mentioned up top, but do we not add `<pytest-filepath>` too?

<img width="862" alt="image" src="https://user-images.githubusercontent.com/31111752/171318661-10f828b0-2e16-41ab-bac0-9536d13a0a10.png">

## astro dev restart

> Stop your Airflow environment, rebuild your Astronomer project into a Docker image, and restart your Airflow environment with the new Docker image.

Reviewer: ```suggestion
Stop your Airflow environment, rebuild your Astro project into a Docker image, and restart your Airflow environment with the new Docker image.
```

> This command can be used to rebuild an Astronomer project and run it locally. For more information, read [Develop and Run a Project Locally](develop-project.md#build-and-run-a-project-locally).

Reviewer: ```suggestion
This command can be used to rebuild an Astro project and run it locally. For more information, read [Develop and Run a Project Locally](develop-project.md#build-and-run-a-project-locally).
```
